src/createTokenBridgePrepareSetWethGatewayTransactionRequest.ts

```diff
@@ -98,6 +99,12 @@ export async function createTokenBridgePrepareSetWethGatewayTransactionRequest({
 }: CreateTokenBridgePrepareRegisterWethGatewayTransactionRequestParams) {
   const chainId = validateParentChain(parentChainPublicClient);
 
+  // ethers providers
+  const parentChainProvider = publicClientToProvider(parentChainPublicClient);
+  const orbitChainProvider = publicClientToProvider(orbitChainPublicClient);
+
+  await registerNewNetwork(parentChainProvider, orbitChainProvider, rollup);
```

Comment:
Is it possible to move this to inside the `getEstimateForSettingGateway` function? I believe networks only need to be registered when calling the estimation method in the SDK, and it might make sense to do the operation there.

Reply:
Also, @spsjvc , what do you think about shifting the responsibility of creating the provider from the publicClient to the createTokenBridge-ethers wrapper? From my point of view, it doesn't really make sense to force prepareTransactionRequest into doing that conversion if it's not really going to use it.